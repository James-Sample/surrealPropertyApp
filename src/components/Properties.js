import React, { useState, useEffect } from "react";
import axios from "axios";
import PropTypes from "prop-types";
import PropertyCard from "./PropertyCard";
import Alert from "./Alert";
import "../styles/properties.css";

const Properties = () => {
  const initialState = {
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const [properties, setProperties] = useState([]);

  const [alert, setAlert] = useState(initialState.alert);

  useEffect(() => {
    axios
      .get("http://localhost:4000/api/v1/PropertyListing/")
      .then(({ data }) => {
        setProperties(data);
      })
      .catch(() => {
        setAlert({
          message: "Unable to retrieve properties",
          isSuccess: false,
        });
      });
  }, []);

  return (
    <div className="properties-page">
      <h2>Properties page</h2>
      <div className="property-cards">
        <Alert message={alert.message} success={alert.isSuccess} />
        {properties.map((property) => (
          <div className="properties">
            <PropertyCard key={property._id} {...property} />
          </div>
        ))}
      </div>
    </div>
  );
};

Properties.propTypes = {
  properties: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      type: PropTypes.string,
      bathrooms: PropTypes.string,
      bedrooms: PropTypes.string,
      price: PropTypes.string,
      city: PropTypes.string,
      email: PropTypes.string,
    })
  ).isRequired,
};
export default Properties;
