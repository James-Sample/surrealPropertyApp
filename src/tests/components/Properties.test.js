import React from "react";
import { render } from "@testing-library/react";
import Properties from "../../components/Properties";

describe("rendering multiple property cards", () => {
  const validProps = [
    {
      title: "2 bed",
      type: "Flat",
      bathrooms: 2,
      bedrooms: 2,
      price: 10000,
      city: "Manchester",
      email: "surreal@estate.com",
    },
    {
      title: "3 bed",
      type: "House",
      bathrooms: 2,
      bedrooms: 3,
      price: 20000,
      city: "Leeds",
      email: "real@estate.com",
    },
  ];
  it("renders the cards", () => {
    const { asFragment } = render(<Properties properties={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  //   it("renders the correct number of cards", () => {
  //     const { asFragment } = render(<Properties properties={validProps} />);
  //     expect()
  //   })
});
