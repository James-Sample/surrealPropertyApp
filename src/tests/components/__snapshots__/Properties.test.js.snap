// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering multiple property cards renders the cards 1`] = `
<DocumentFragment>
  <div
    class="properties-page"
  >
    <h2>
      Properties page
    </h2>
    <div
      class="property-cards"
    />
  </div>
</DocumentFragment>
`;
